src/execution/subscribe.ts

```diff
@@ -170,7 +131,45 @@ export function createSourceEventStream(
   variableValues?: Maybe<{ readonly [variable: string]: unknown }>,
   operationName?: Maybe<string>,
   subscribeFieldResolver?: Maybe<GraphQLFieldResolver<any, any>>,
+): PromiseOrValue<AsyncIterable<unknown> | ExecutionResult>;
+export function createSourceEventStream(
+  argsOrSchema: ExecutionArgs | GraphQLSchema,
+  document?: DocumentNode,
+  rootValue?: unknown,
+  contextValue?: unknown,
+  variableValues?: Maybe<{ readonly [variable: string]: unknown }>,
+  operationName?: Maybe<string>,
+  subscribeFieldResolver?: Maybe<GraphQLFieldResolver<any, any>>,
 ): PromiseOrValue<AsyncIterable<unknown> | ExecutionResult> {
+  if (isSchema(argsOrSchema)) {
+    invariant(document != null, 'Must provide document.');
```

Comment:
Do you still need this `invariant` if you make separate TS signatures?

Reply:
In general, I think his approach is cleaner than doing "Impl" function.

Reply:
I tried that approach but couldn’t get rid of invariant! @n1ru4l  any ideas?

Reply:
I think unfortunately a TS limitation see [c39e0f3](https://github.com/graphql/graphql-js/pull/3634/commits/c39e0f350feb7f6be8b67f4da98885ced7991ca5)